src/common/PresenceStatusIndicator.js

```diff
@@ -38,18 +48,28 @@ type Props = {|
  * * gray if 'offline'
  *
  * @prop [style] - Style object for additional customization.
- * @prop [presence] - UserPresence object used to determine the status from.
+ * @prop [email] - email of the user whose status we are showing.
  * @prop hideIfOffline - Do not render for 'offline' state.
  */
-export default class PresenceStatusIndicator extends PureComponent<Props> {
+class PresenceStatusIndicator extends PureComponent<Props> {
   render() {
-    const { presence, style, hideIfOffline } = this.props;
+    const { email, presence, style, hideIfOffline, usersByEmail, userStatus } = this.props;
+
+    const userPresence = presence[email];
 
-    if (!presence || !presence.aggregated) {
+    if (!userPresence || !userPresence.aggregated) {
       return null;
     }
 
-    const status = statusFromPresence(presence);
+    const user = usersByEmail[email];
+
+    if (!user) {
+      return null;
+    }
+
+    const awayStatus = userStatus[user.user_id];
+
+    const status = awayStatus && awayStatus.away ? 'offline' : statusFromPresence(userPresence);
```

Comment:
Hmm. I think I'd rather this logic be encapsulated in its own function or method somewhere.

A minimal change would be to put it in a method on the component -- so like
```
render() {
  const status = this.getStatus();
  if (status === null) {
    return null;
  }

  const { style, hideIfOffline } = this.props;
  // if (hideIfOffline ... etc., etc.
```

and that would I think help a lot in separating the logic for the sake of reading and understanding this code.

Ultimately I'd want the substance of that method, especially the `awayStatus.away ? 'offline' : statusFromPresence(...)` part, to be in a selector or similar function somewhere, outside of display code. But I'm OK merging without that.